src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -49,14 +50,22 @@ const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValida
   const value = useValue(uischema);
   const { loading } = useWidgetContext();
   const onChangeHandler = useOnChange(uischema);
-  const label = getTranslation(uischema.translations!, 'label');
   const { translations = [], focus, required } = uischema;
   const {
     attributes,
     inputMeta: { name },
   } = uischema.options;
+  const label = getTranslation(translations, 'label');
+  const hint = getTranslation(translations, 'hint');
+  const explain = getTranslation(translations, 'bottomExplain');
   const focusRef = useAutoFocus<HTMLInputElement>(focus);
   const hasErrors = typeof errors !== 'undefined';
+  const hintEleId = `${name}-hint`;
+  const explainEleId = `${name}-explain`;
+  const hintAndExplainIds = `${explain ? ` ${explainEleId} ` : ''}${hint ? ` ${hintEleId}` : ''}`;
+  const updatedDescribedByIds = typeof describedByIds !== 'undefined'
```

Comment:
Do you need to check so specifically that it's `undefined`?

```suggestion
  const updatedDescribedByIds = describedByIds
```

Reply:
In this case I believe so because if I don't it will concat `undefined` as a string to the hint/explain describedBy IDs 

```
const updatedDescribedByIds = typeof describedByIds !== 'undefined'
    ? `${describedByIds}${hintAndExplainIds}`
    : `${hintAndExplainIds}`;
```

So if I dont check if `describedByIds` is actually undefined, it will result in `updatedDescribedByIds = 'undefined credentials.passcode-hint';

Reply:
I ultimately extracted this into a util function since this logic was repeated in two components. see: `src/util/buildInputDescribedByValue.ts` and its corresponding tests.